File: musicApi/src/main/java/com/svalero/musicApi/domain/Album.java
package com.svalero.musicApi.domain;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import jakarta.validation.constraints.Min;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity(name="album")
public class Album {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(nullable = false, length = 150)
    private String title;

    @Column(name = "release_date")
    private LocalDate releaseDate;

    @Column(length = 100)
    private String front;

    @Column(length = 70)
    private String genre;

    @Column
    @Min(1)
    private int tracks;

    @Column
    @Min(0)
    private double duration;

    @Column(length = 70)
    private String label;

    @Column
    @ColumnDefault("false")
    private boolean platinum;

    @ManyToOne
    @JoinColumn(name = "artist_id")
    //Avoids infinite JSON serialisation, as team-player is related and enters a loop
    @JsonManagedReference(value = "artist_albums")
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Artist artist;
}

File: musicPhp/models/Album.php
<?php

namespace models;

class Album {

    public $id;
    //public $name;
    public $title;
    public $releaseDate;
    public $genre;
    public $front;
    public $tracks;
    public $duration;
    public $label;
    public $platinum;


    public function __construct($data) {
        $this->id = $data["id"];
        //$this->name = $data["name"];
        $this->title = $data["title"];
        $this->releaseDate = $data["releaseDate"];
        $this->genre = $data["genre"];
        $this->front = $data["front"];
        $this->tracks = $data["tracks"];
        $this->duration = $data["duration"];
        $this->label = $data["label"];
        $this->platinum = $data["platinum"];

    }

}
?>